node/src/actors/chain_manager/mod.rs

```diff
@@ -763,11 +763,20 @@ impl ChainManager {
                             self.persist_data_requests(ctx, to_be_stored);
                         }
 
-                        let _reveals = self
+                        let reveals = self
                             .chain_state
                             .data_request_pool
                             .update_data_request_stages();
 
+                        for reveal in reveals {
+                            // Send AddTransaction message to self
+                            // And broadcast it to all of peers
+                            ctx.address().do_send(AddTransaction {
```

Comment:
If the node is not in `Synced` state, `AddTransaction` will return `ChainManagerError::NotSynced`.